### ISSUE TITLE
Wiki page missing, yet its content shows up in wiki search...

### ISSUE THREAD

**Issue by cliveb** [fedwiki/wiki-client#284]
Any idea why a wiki page goes missing?

http://clive.tries.fed.wiki/view/recent-changes/view/teradata-v-sap

The page content shows up in search. 

I am using a Chromebox, as I normally do writing on wiki, with the latest ChromeOS.
![page missing found in search](https://user-images.githubusercontent.com/733870/152479884-611322e0-d949-4f0e-8f1c-9c3a561d9bdf.png)

I write this page from about 1 am to 3 am PST Feb 3, 2022.

Thanks for any pointers.

**Comment by cliveb** [comment]
Update. I found the page from the wiki search, by clicking the color wiki icon square in the browser (not the blue link in the browser). The page loaded, in the open window, I forked the page, but the only the page paragraph in the observable in the wiki browser forked. 

Is this my user error. 

I noticed another inconsistency (not 100 percent sure) I found I could fork the page without signing to my wiki account. Again the fork was only the paragraph seen in the wiki search browser preview. 

Any chance the wiki server DB has reached its allocated size limit, and is now writing beyond memory? A similar pattern infamously happened to FourSquare whose database had disk space but the space was not allocated in DB, which caused the index to be searchable but the document pages were not found (I think 4sq was on mongodb)

.  
![forks only wiki search paragraph](https://user-images.githubusercontent.com/733870/152644162-2f681b8f-ef82-44f6-b9ff-aef3c83b2d9f.png)

**Comment by paul90** [comment]
Interesting...

Looking at the list of pages, I see the slug is `teradata-v-sap-`. So, I suspect the title has a trailing space. On the "Deep Thoughts" page the link as a trailing space in the link.

You ask, "User error?" - well that may be, but it uncovers some inconsistency in how spaces in links, titles, and slugs are trimmed. Which will need investigating.

In the sitemap we see,
```
{
  "slug":"teradata-v-sap-",
  "title":"Teradata v SAP",
  "date":1643884690759,
  "synopsis":"A good example of a So...186/603 docket]"
}
```
which suggests that the trailing space in the title has been trimmed off at some point. Which creates a mismatch between the slug generated from the title, and the slug used to save the page - which almost certainly explains the other strangeness.

**Comment by cliveb** [comment]
Yes. Even wiki page edits and page forks are all malformed. The trailing space appears to cause all the strangeness. 
![Strangeness](https://user-images.githubusercontent.com/733870/152663862-27de7fd4-f4b7-4ecb-90ea-df8ce7a6549a.png)

**Comment by cliveb** [comment]
I safely copied the trailing space page content to a new page. Key to find the orphaned page for anyone else who stumbles on this is use wiki search to find the page in the browser and click your colored wiki square in the search not the blue page link (I think). 
![copy trailing space page](https://user-images.githubusercontent.com/733870/152664009-fc888af4-fe19-4ad7-9d24-bdbc15c7a0dc.png)

**Comment by WardCunningham** [comment]
I feel like in any situation where the spaces would be visible we would be justified in honoring them.

A case could be made that trailing spaces are invisible when typing into the Create New Page form from the hamburger menu and would be improved by trimming input.

**Comment by WardCunningham** [comment]
Noted elsewhere:
http://hive.dreyeck.ch/view/slug
<img width="450" alt="image" src="https://user-images.githubusercontent.com/12127/233537749-e3f5bf48-b6d4-4b1b-96ce-fa70ba585bb8.png">

**Comment by cliveb** [comment]
I feel trailing spaces are inadvertent.
